crates/types/src/string.rs

```diff
@@ -22,6 +22,17 @@ pub struct String {
     pub(super) size: usize,
 }
 
+/// A builder that can be used to efficiently build a [`nvim_oxi::String`](String).
+#[repr(C)]
+pub struct StringBuilder {
```

Comment:
I'm not sure this should be part of the public API. Can we make it `pub(crate)`?

Reply:
It probably can't be `pub(crate)` because the `string!` macro uses it. The macro can't expand to anything that is private.

![screenshot](https://github.com/user-attachments/assets/3ab777c3-9233-4c21-9da6-08552ce39113)

I'm not sure if there is another way to write the macro. I think making this public is fine, people might want to build strings at runtime, and the `string!` macro might not be useful for that.

Reply:
Come to think of it, we can probably make a `format` function that uses the `StringBuilder` internally. But as i said, making it public is useful to users. Just as `smol_str`'s builder is also public.